cli/azd/pkg/ext/models.go

```diff
@@ -125,7 +125,10 @@ func (hc *HookConfig) validate() error {
 		hc.Shell = scriptType
 	}
 
-	hc.validated = true
+	// stored validated, except for inline script which needs to be regenerated per-usage
+	if hc.location != ScriptLocationInline {
```

Comment:
This was a newly uncovered flaw that may have surfaced some other place.

1. Inline script is currently generated as part of `validate`
2. After an inline script runs, we delete the temp script.
3. This means if the event that the inline script subscribes to, was retriggered, we would skip the `validate` step and end up running the script without first generating the inline script.

I can also move it out of the `validate`, but it ends up being a bigger change. If people are OK with it, I'd prefer post-release cleanup.